### DISCUSSION TITLE
SMTP Settings for Mailgun

### DISCUSSION THREAD

**Discussion by dwayneem** [mastodon/mastodon#21714] [Q&A]
I have:
SMTP server: smtp.mailgun.org
SMTP username: postmaster@mg.supportdifferent.com
SMTP password: 
SMTP authentication: plain
SMTP OpenSSL verify mode: peer
Enable STARTTLS: auto

But get the error:
E-mail could not be sent with this configuration, try again.
execution expired

I found a discussion that said that "SMTP OpenSSL verify mode" should be "peer" and not "none" but no mention of the "Enable STARTTLS" option. I have tried all three values for the "Enable STARTTLS" option, but still get the same error message when it tries to send a sample email.

**Comment by davetroy** [comment]
I've used these settings with both Mailgun and Sendgrid successfully.

```
SMTP_PORT=587
SMTP_SERVER=smtp.sendgrid.com
SMTP_LOGIN=<YOUR_LOGIN>
SMTP_PASSWORD=<YOUR_PASSWORD>
SMTP_AUTH_METHOD=plain
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_FROM_ADDRESS='ToadSocial <notifications@toad.social>'
```

If this doesn't work, your login or password is probably incorrect. Good luck!

**Reply by galfridus** [reply]
From https://peterbabic.dev/blog/setting-up-smtp-in-mastodon/, try:

SMTP_SERVER=smtp.example.com
SMTP_PORT=465
SMTP_LOGIN=mastodon@peterbabic.dev
SMTP_PASSWORD=very-strong-passphrase-here
SMTP_FROM_ADDRESS=mastodon@peterbabic.dev
SMTP_SSL=true
SMTP_ENABLE_STARTTLS_AUTO=false
SMTP_AUTH_METHOD=plain
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_DELIVERY_METHOD=smtp

Worked for me after a lot of frustration.
